src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -421,7 +423,7 @@ def __init__(
         event: Event | GroupEvent | None = None,
         tags: set[str] | None = None,
         identity: RpcIdentity | None = None,
-        actions: Sequence[MessageAction] | None = None,
+        actions: Sequence[MessageAction | BlockKitMessageAction] | None = None,
```

Comment:
what is the difference between `MessageAction` and `BlockKitMessageAction`? should we always be using one over the other if we don't support non-block kit anymore?

Reply:
ideally we should delete `MessageAction`, but i saw some places in our logic where we create a `MessageAction` payload to send to slack, so i want to slowly remove it so make sure we don't break changes since ff this is hard

Reply:
Yeah, as I mentioned [here](https://github.com/getsentry/sentry/pull/77444#discussion_r1757607830) it seems like `BlockKitMessageAction` ought to extend `MessageAction` (assuming we need both).

The "TODO" comment on `MessageAction` suggests that its bottom four attributes (including `block_id`, which is confusing) ought to belong to some other subclass. But I agree with delaying that sort of refactoring for later.

I'm okay with using `MessageAction | BlockKitMessageAction` for type hints temporarily, but would consider changing `BlockKitMessageAction` to extend `MessageAction` instead.